frontend/src/pages/modelServing/screens/metrics/EnsureMetricsAvailable.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Bullseye, PageSectionVariants, Spinner } from '@patternfly/react-core';
+import UnauthorizedError from '~/pages/UnauthorizedError';
+import {
+  ModelMetricType,
+  ModelServingMetricsContext,
+  ServerMetricType,
+} from '~/pages/modelServing/screens/metrics/ModelServingMetricsContext';
+
+const DEFAULT_ACCESS_DOMAIN = 'metrics';
+
+type EnsureMetricsAvailableProps = {
+  children: React.ReactElement;
+  metrics: ModelMetricType[] | ServerMetricType[];
+  accessDomain?: React.ReactNode;
+};
+
+const EnsureMetricsAvailable: React.FC<EnsureMetricsAvailableProps> = ({
+  children,
+  metrics,
+  /** Resource name to be displayed if access denied, e.g. 'server metrics', 'bias metrics', etc. */
+  accessDomain = DEFAULT_ACCESS_DOMAIN,
+}) => {
+  const { data } = React.useContext(ModelServingMetricsContext);
+  let error;
+  let readyCount = 0;
+
+  metrics.forEach((metric) => {
+    if (data[metric].error) {
+      error = data[metric].error;
+    }
+    data[metric].loaded && readyCount++;
+  });
```

Comment:
Nit, this is a reduce

Reply:
Hmmm I didn't reach for reduce as this is reducing an array to 2 separate values - I guess I could have just use a tuple type for the accumulator.